src/test/PoolDonateTest.sol

```diff
@@ -52,13 +55,17 @@ contract PoolDonateTest is PoolTestBase, Test {
         (,, uint256 reserveAfter0, int256 deltaAfter0) = _fetchBalances(data.key.currency0, data.sender);
         (,, uint256 reserveAfter1, int256 deltaAfter1) = _fetchBalances(data.key.currency1, data.sender);
 
-        assertEq(reserveBefore0, reserveAfter0);
-        assertEq(reserveBefore1, reserveAfter1);
-        assertEq(deltaAfter0, int256(data.amount0));
-        assertEq(deltaAfter1, int256(data.amount1));
+        if (!data.key.hooks.shouldAccessLock()) {
+            assertEq(reserveBefore0, reserveAfter0);
+            assertEq(reserveBefore1, reserveAfter1);
+            assertEq(deltaAfter0, int256(data.amount0));
+            assertEq(deltaAfter1, int256(data.amount1));
+        }
```

Comment:
So we're not checking reserves and deltas for access lock hooks since the hook will be modifying the original sender's lock, are we pushing the assertions into the test file instead?

Reply:
Yeah the assertions I was testing for AccessLock are after the PoolManager calls 

like here https://github.com/Uniswap/v4-core/blob/access-lock/test/AccessLock.t.sol#L132 

I don't check reserves but I think these assertions make sense? Lmk if you think theres something else I should check though